src/test/java/org/ice4j/message/ChannelDataTest.java

```diff
@@ -0,0 +1,85 @@
+package org.ice4j.message;
+
+import org.ice4j.StunException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.ice4j.message.ChannelData.*;
+import static org.junit.Assert.*;
+
+public class ChannelDataTest {
+
+  @Rule public final ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void testDecode() throws StunException {
```

Comment:
I don't think you need the `throws StunException` here

Reply:
I think this is ok. If I remove it, I get an error:

> Error:(19, 19) java: unreported exception org.ice4j.StunException; must be caught or declared to be thrown

and if I "expect" an exception by adding a "thrown" statement, I get an error about the fact an exception is not thrown:

> java.lang.AssertionError: Expected test to throw an instance of org.ice4j.StunException

It looks (to me) like the "throws" is needed because the tested method is declared with a "throws"(?)